mantle/kola/README-kola-ext.md

```diff
@@ -10,28 +10,38 @@ repositories, and allow these projects to target Fedora CoreOS
 in e.g. their own CI.  And we also want to run unmodified
 upstream tests, *without rebuilding* the project.
 
-Using kola run-ext-bin
+Using kola run -E/--exttest
 ===
 
-The `kola run-ext-bin` is one way to accomplish this.  The
-core idea is to express your test suite as a binary (plus an optional
-directory of dependencies) that will be uploaded to a CoreOS
-system and run as a systemd unit.
+The `--exttest` (`-E`) argument to `kola run` one way to accomplish this; you
+provide the path to a directory which will be traversed recursively.
 
-Currently this systemd unit runs with full privileges - tests
+The core idea is to express your test suite as a binary (plus an optional
+directory of dependencies named `deps`) that will be uploaded to a CoreOS
+system and run as a systemd unit, along with an optional Ignition config
+named `config.ign`.
+
+Concretely then, an external test directory can have the following content:
+
+- `config.ign` (optional): Ignition config provided
+- `deps` (optional): Directory (or symlink to dir): Will be uploaded and available as `KOLA_EXT_DATA`
+- one or more executables: Each executable is its own test, run independently
+  with the Ignition config and/or dependencies provided.
+
```

Comment:
I think something referencing the transformation could be helpful. Something like:

```suggestion

*Note*: When adding tests in this way the test prefix comes from the basename of the directory and the test name is the basename of the binary itself. As an example `/path/to/externaltest/tests/mytest` where `-E /path/to/externaltest/` was used would be available in the kola test suite as `externaltest.tests.mytest`.
```

Reply:
I adapted a variant of this, thanks!